tests/unit/plugins/kubernetes/kubernetes_pod_executor_test.py

```diff
@@ -318,6 +318,55 @@ def test_pending_event_processing_loop_processes_remaining_events_after_stop(k8s
     assert k8s_executor.pending_events.qsize() == 0
 
 
+@mock.patch(
+    "task_processing.plugins.kubernetes.kubernetes_pod_executor.REFRESH_EXECUTOR_STATE_THREAD_GRACE",  # noqa
+    1,
+)
+@mock.patch(
+    "task_processing.plugins.kubernetes.kubernetes_pod_executor.REFRESH_EXECUTOR_STATE_THREAD_INTERVAL",  # noqa
+    1,
+)
+def test_reconcile_task_loop_reconcile_existing_pods_only(k8s_executor, mock_task_configs):
+    mock_pods = []
+    test_phases = ['Unknown', 'Succeeded', 'Failed', 'Running']
+    task_metadata = {}
+
+    for taskconf, phase in zip(mock_task_configs, test_phases):
+        mock_pod = mock.Mock(spec=V1Pod)
+        mock_pod.metadata.name = taskconf.pod_name
+        mock_pod.status.phase = phase
+        mock_pod.status.host_ip = '1.2.3.4'
+        mock_pod.spec.node_name = 'kubenode'
+        mock_pods.append(mock_pod)
+        task_metadata[taskconf.pod_name] = KubernetesTaskMetadata(
+            task_config=taskconf,
+            task_state=KubernetesTaskState.TASK_RUNNING,
+            task_state_history=v(),
+        )
+    # Keep only 3 task_configs in task_metadata
+    task_metadata.popitem()
```

Comment:
can we generate just the 3 tasks to minimize the logic in this test?

Reply:
This is actually intentional, since we aren't testing the actual reconciliation function I wanted to see if we had 4 V1Pods from kubernetes but only 3 in task_metadata it would reconcile those 3 tasks and ignore the 4th pod

Reply:
ah gotcha - might be good to explicitly document that in this comment :)

Reply:
added some additional comments :)